test/unit/gateways/mercury_test.rb

```diff
@@ -63,12 +63,39 @@ def test_successful_refund
     assert refund.test?
   end
 
-  def test_add_swipe_data_with_credit_card
-    @credit_card.track_data = "data"
+  def test_cardpresent_with_track_1_data
+    track_data = "%B4003000123456781^LONGSEN/L. ^15121200000000000000123?"
+    @credit_card.track_data = track_data
     response = stub_comms do
       @gateway.purchase(@amount, @credit_card, @options)
     end.check_request do |endpoint, data, headers|
-      assert_match(/Track1>data</, data)
+      assert_match(/<Track1>#{Regexp.escape(track_data)}<\/Track1>/, data)
+    end.respond_with(successful_purchase_response)
+
+    assert_instance_of Response, response
+    assert_success response
+  end
+
+  def test_cardpresent_with_track_2_data
+    track_data = ";5413330089010608=2512101097750213?"
+    @credit_card.track_data = track_data
+    response = stub_comms do
+      @gateway.purchase(@amount, @credit_card, @options)
+    end.check_request do |endpoint, data, headers|
+      assert_match(/<Track2>#{Regexp.escape(track_data)}<\/Track2>/, data)
+    end.respond_with(successful_purchase_response)
+
+    assert_instance_of Response, response
+    assert_success response
+  end
+
+  def test_cardpresent_with_bad_track_data_uses_track1_endpoint
```

Comment:
Yep, bit of a misnomer on the test name. Something like `test_bad_track_data_in_track_1` should suffice.